service/routes.go

```diff
@@ -15,14 +18,41 @@ func (svc *Service) buildRoutes(ctx context.Context) {
 	if svc.HealthCheck != nil {
 		svc.Router.Handle("/health", http.HandlerFunc(svc.HealthCheck.Handler))
 	}
-	svc.publicEndpoints(ctx)
+
+	if svc.Config.EnablePrivateEndpoints {
+		svc.privateEndpoints(ctx)
+	} else {
+		svc.publicEndpoints(ctx)
+	}
+
 }
 
 func (svc *Service) publicEndpoints(ctx context.Context) {
 	log.Info(ctx, "enabling public endpoints")
 
 	// Routes
-	populationTypes := handler.NewPopulationTypes(svc.responder, svc.cantabularClient)
+	populationTypes := handler.NewPopulationTypes(svc.responder, svc.cantabularClient, svc.identityClient, svc.DatasetAPIClient)
 	svc.Router.Get("/population-types", populationTypes.Get)
 	svc.Router.Get("/population-types/{population-type}/area-types", populationTypes.GetAreaTypes)
 }
+
+func (svc *Service) privateEndpoints(ctx context.Context) {
+	log.Info(ctx, "enabling private endpoints")
+
+	populationTypes := handler.NewPopulationTypes(svc.responder, svc.cantabularClient, svc.identityClient, svc.DatasetAPIClient)
+
+	r := chi.NewRouter()
+
+	permissions := middleware.NewPermissions(svc.Config.ZebedeeURL, svc.Config.EnablePermissionsAuth)
+	checkIdentity := dphandlers.IdentityWithHTTPClient(svc.identityClient)
+
+	r.Use(checkIdentity)
+	r.Use(middleware.LogIdentity())
+	r.Use(permissions.Require(auth.Permissions{Read: true}))
+
+	r.Get("/population-types", populationTypes.Get)
+	r.Get("/population-types/{population-type}/area-types", populationTypes.GetAreaTypes)
+
+	// added from cant filt flex
```

Comment:
Not a good comment to leave hanging about, it just says 'I copied this from somewhere else, I don't actually know what it does'